### ISSUE TITLE
Flycheck doesn't show errors for Python mode with ty lsp

### ISSUE THREAD

**Issue by FabriceSalvaire** [flycheck/flycheck-eglot#15]
It looks like #11.
Flycheck shows any errors, **ty** reports errors, and it works fine with **pyright** lsp.

Any clue ? Packages messed ???

I found in the events:
```
[jsonrpc] e[04:02:27.207] --> textDocument/diagnostic[2]

{
    "jsonrpc": "2.0",
    "id": 2,
    "method": "textDocument/diagnostic",
    "params": {
        "textDocument": {
            "uri": "file:///home/.../test-color.py"
        }
    }
}

[jsonrpc] e[04:02:27.422] <-- textDocument/diagnostic[2]

{
    "jsonrpc": "2.0",
    "id": 2,
    "result": {
        "items": [
            {
                "code": "invalid-argument-type",
                "codeDescription": {
                    "href": "https://ty.dev/rules#invalid-argument-type"
                },
                "data": null,
                "message": "Argument to function `to_float_rgb` is incorrect: Expected `Int3`, found `Literal[128]`",
                "range": {
                    "end": {
                        "character": 22,
                        "line": 8
                    },
                    "start": {
                        "character": 19,
                        "line": 8
                    }
                },
                "severity": 1,
                "source": "ty"
            },
...
            {
                "code": "invalid-argument-type",
                "codeDescription": {
                    "href": "https://ty.dev/rules#invalid-argument-type"
                },
                "data": null,
                "message": "Argument to function `foreground` is incorrect: Expected `RGBColor`, found `Int3`",
                "range": {
                    "end": {
                        "character": 38,
                        "line": 35
                    },
                    "start": {
                        "character": 31,
                        "line": 35
                    }
                },
                "severity": 1,
                "source": "ty"
            }
        ],
        "kind": "full",
        "resultId": "407a3c4ee3afb413"
    }
}
```

**Comment by FabriceSalvaire** [comment]
@intramurz Thanks for your answer !

I rechecked that
- `flycheck / eglot` **works** with `pyright` lsp
- `flymake` **works** with `ty` lsp
- `flycheck / eglot` **doesn't work** with `ty` lsp
- `flycheck-current-errors` and `eglot--pushed-diagnostics` are **nil**
- `eglot--pulled-diagnostics` is **not nil**
- `ty` is mentioned in Doom doc, but Doom is currently broken... so any clue from there
- I had the intuition to compare jsonrpc between pyright and ty : **it is a bit different !!!**
- **TIPS for newbies** `eglot-events-buffer` and `json-reformat-region` are useful to debug the jsonrpc

**Pyright is**
```
[jsonrpc] e[04:15:39.779] <-- textDocument/publishDiagnostics

{
    "jsonrpc": "2.0",
    "method": "textDocument/publishDiagnostics",
    "params": {
        "uri": "file:///home/fabrice/home/developpement/python/vt100_toolkit/test-color.py",
        "version": 0,
        "diagnostics": [
            {
                "range": {
                    "start": {
                        "line": 8,
                        "character": 19
                    },
                    "end": {
                        "line": 8,
                        "character": 22
                    }
                },
                "message": "Argument of type \"Literal[128]\" cannot be assigned to parameter \"rgb\" of type \"Int3\" in function \"to_float_rgb\"\n  \"Literal[128]\" is not assignable to \"tuple[int, int, int]\"",
                "severity": 1,
                "code": "reportArgumentType",
                "source": "Pyright",
                "codeDescription": {
                    "href": "https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType"
                }
            },
            ...
            {
                "range": {
                    "start": {
                        "line": 19,
                        "character": 22
                    },
                    "end": {
                        "line": 19,
                        "character": 25
                    }
                },
                "message": "Argument of type \"Literal[234]\" cannot be assigned to parameter \"rgb\" of type \"Int3\" in function \"to_hex\"\n  \"Literal[234]\" is not assignable to \"tuple[int, int, int]\"",
                "severity": 1,
                "code": "reportArgumentType",
                "source": "Pyright",
                "codeDescription": {
                    "href": "https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType"
                }
            },
            {
                "range": {
                    "start": {
                        "line": 35,
                        "character": 31
                    },
                    "end": {
                        "line": 35,
                        "character": 38
                    }
                },
                "message": "Argument of type \"Int3\" cannot be assigned to parameter \"rgb\" of type \"RGBColor\" in function \"foreground\"\n  \"tuple[int, int, int]\" is not assignable to \"list[int]\"",
                "severity": 1,
                "code": "reportArgumentType",
                "source": "Pyright",
                "codeDescription": {
                    "href": "https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType"
                }
            }
        ]
    }
}

```

**Comment by FabriceSalvaire** [comment]
I am looking lisp code...
`eglot--pushed-diagnostics` is also nil for `flymake`

**Comment by FabriceSalvaire** [comment]
hum ok the issue should be in https://github.com/joaotavora/eglot
I learnt eglot manage flymake.

**Comment by FabriceSalvaire** [comment]
@intramurz Are you aware of this function ???

I looked for `textDocument/diagnostic` in my elpa and found it

```
eglot-1.21/eglot.el
3419:              :textDocument/diagnostic
3436:              :hint :textDocument/diagnostic)))))
3474:       ;; Either a textDocument/diagnostics response specifically told
```

```
(cl-defun eglot--flymake-pull (&aux (server (eglot--current-server-or-lose))
                                    (origin (current-buffer)))
  "Pull diagnostics from server, for all managed buffers.
When response arrives call registered `eglot--flymake-report-fn'."
  (cl-flet
      ((pull-for (buf &optional then)
         (with-current-buffer buf
           (let ((version eglot--docver)
                 (prev-result-id (caddr eglot--pulled-diagnostics)))
             (eglot--async-request
              server
              :textDocument/diagnostic
              (append
               `(:textDocument ,(eglot--TextDocumentIdentifier)
                               ,@(when prev-result-id
                                   `(:previousResultId ,prev-result-id))))
              :success-fn
              (eglot--lambda ((DocumentDiagnosticReport) kind items resultId)
                (eglot--when-live-buffer buf
                  (pcase kind
                    ("full"
                     (setq eglot--pulled-diagnostics
                           (list items version resultId))
                     (eglot--flymake-report-push+pulled :force t))
                    ("unchanged"
                     (when (eq buf origin)
                       (eglot--flymake-report-1 nil :stay :force t)))))
                (when then (funcall then)))
              :hint :textDocument/diagnostic)))))
    ;; JT@2025-12-15: No known server yet supports "relatedDocuments" so
    ;; the only way we have to get related diagnostics is to explicitly
    ;; request them of all open documents.  Moreover, experience has
    ;; shown this needs to happen after the 'origin''s response.
    (pull-for origin
              (unless (zerop eglot--docver)
                (lambda ()
                  (mapc #'pull-for
                        (remove origin (eglot--managed-buffers server))))))))
```

**Comment by FabriceSalvaire** [comment]
I did further debugging and I have diags in **Messages**
Is there something broken in flyspell ???

I also noticed eglot/ty inject information like "types" in the buffer.
This is unusual and annoying, a kind of ghost faces.
Where is it documented ???
How to disable that ???

My debugging
```
(defun flycheck-eglot--report-eglot-diagnostics (diags &rest _)
  "Report function for the `eglot-flymake-backend'.
DIAGS is the Eglot diagnostics list in Flymake format."
  (cl-flet
      ((diag-to-err (diag)
         ;; Translate flymake to flycheck
         ;; (message (format "diag %S" diag))
         (message (format "diag buffer %S / beg %S end %S / level %S / text %S"
                          (flymake-diagnostic-buffer diag)
                          (flymake-diagnostic-beg diag) ; POS
                          (flymake-diagnostic-end diag)
                          (flycheck-eglot--get-error-level diag) ; LEVEL
                          (flymake-diagnostic-text diag)  ; MESSAGE
                          ))
                    (with-current-buffer (flymake-diagnostic-buffer diag)
                        (flycheck-error-new-at-pos
                       (flymake-diagnostic-beg diag) ; POS
                       (flycheck-eglot--get-error-level diag) ; LEVEL
                       (flymake-diagnostic-text diag)  ; MESSAGE
                       :end-pos (flymake-diagnostic-end diag)
                       :checker 'eglot-check
                       :buffer (current-buffer)
                       :filename (buffer-file-name)))))

    (message "in flycheck-eglot--report-eglot-diagnostics")
    (setq flycheck-eglot--current-errors
          (mapcar #'diag-to-err diags))
    (message "set flycheck-eglot--report-eglot-diagnostics %S" flycheck-eglot--current-errors) ; structure is there !
    (flycheck-buffer-automatically '(idle-change new-line))))
```

**Comment by FabriceSalvaire** [comment]
@intramurz Sorry I forgot to answer. Your fix works ! Many thanks !

However, I had to use Rassumfrassum LSP proxy, because ty and ruff are two independent LSP...
I don't know if your fix is "used" by rassumfrassum ???

Unfortunately, it is now broken since ty v0.16, I opened https://github.com/astral-sh/ty/issues/3067
since I don't know what they changed

**Comment by intramurz** [comment]
Thank you for the kind feedback.

It so happened that almost simultaneously with your issue, I received a [complaint](https://github.com/flycheck/flycheck-eglot/issues/16) about `flycheck-eglot` not working with _rassumfrassum_. The issue was specifically about the same _rassumfrassum_+_ty_+_ruff_ configuration (which `eglot` offers by default for Python if all components are installed). In the end, both with _ty_ alone and with _rassumfrassum_ and _ruff_, everything worked with minimal issues (problems with window margins; something needs to be done, but that's definitely not `flycheck-eglot`'s responsibility).

I updated _ty_ (0.0.23) and _ruff_ (0.15.6) today, and everything works as before. `eglot`'s settings are more or less default, without any tricky configuration. Could it be that the problem is specific to your environment? I haven't been doing any serious Python development lately, so I might be behind the times on this issue, but maybe you could try a simple configuration on a simple toy project first (one code file, no venv)?

Good luck.
